### ISSUE TITLE
Trend Metrics timezone offset off when using other than UTC app.timezone config

### ISSUE THREAD

**Issue by tumainimosha** [laravel/nova-issues#2626]
- Laravel Version: 7.11.0
- Nova Version: 3.4.0
- PHP Version: 7.4.6
- Database Driver & Version: PostgreSQL 11.5 running on AWS RDS
- Operating System and Version: Ubuntu 18.04 running on AWS EC2.
- Browser type and version: Chrome 81
- Reproduction Repository: https://github.com/tumainimosha/nova-issue-demo

### Description:

Related to closed issues #1136 and #1827 

When using other than UTC timezone for config of app.timezone default laravel eloquent behaviour is date fields are stored in db with same timezone as app timezone, and not as UTC, [refer here](https://github.com/laravel/framework/issues/1841)

In trend metrics calculation, the nova calculates a time offset to be applied to date columns, to be presented in user's timezone. However, the offset calculation assumes database stored time is in UTC and calculates offset as hour difference of user's timezone from UTC.

See nova source `src/Metrics/TrendDateExpression.php` at function `offset()`

```php
public function offset()
{
    if ($this->timezone) {
        return (new DateTime(Chronos::now()->format('Y-m-d H:i:s'), new DateTimeZone($this->timezone)))->getOffset() / 60 / 60;
    }

    return 0;
}
```

However, 
If your app uses a different timezone then the offset calculation for database stored times needs to be calculated against your app timezone and not against UTC.

See below suggested changes to `src/Metrics/TrendDateExpression.php` 

```php
public function offset()
{
    if ($this->timezone) {
        $userTimezoneOffset = (new DateTime(Chronos::now()->format('Y-m-d H:i:s'), new DateTimeZone($this->timezone)))->getOffset() / 60 / 60;

        // Take into account system timezone thats already factored in database stored time
        $dbTimezone = config('app.timezone');
        $dbTimezoneOffset = (new DateTime(Chronos::now()->format('Y-m-d H:i:s'), new DateTimeZone($dbTimezone)))->getOffset() / 60 / 60;

        return $userTimezoneOffset - $dbTimezoneOffset;
    }

    return 0;
}
```

### Steps To Reproduce:

1. Set Laravel config `app.timezone` to a value other than UTC (+/-) eg: Africa/Dar_es_Salaam

2. Seed a sample db table (eg: Users) with times spread created_at spread over 24hrs interval from start to end of day.

```php
$startTime = \Carbon\Carbon::yesterday()->startOfDay();
$endTime = \Carbon\Carbon::yesterday()->endOfDay();

// Will create 24 entries space by an hour for yesterdays date
$time = $startTime->clone();
while ($time->lessThan($endTime)) {
    $user = factory(\App\User::class)->make();
    $user->created_at = $time;
    $user->updated_at = $time;
    $user->save();

    $time->addHour();
}
```

3. Create a simple trend metric to display the list of users

```php
public function calculate(NovaRequest $request)
{
    return $this->countByDays($request, User::class);
}
```

4. The screenshot below shows result from step 1 to 4 on a fresh laravel nova install on a new project.
<img width="788" alt="Screenshot 2020-05-19 at 01 39 38" src="https://user-images.githubusercontent.com/7481490/82266217-c6918080-9971-11ea-8fcc-fcae4506ddca.png">

Expected result was for all 24 users created to be counted towards yesterdays date, instead 21 are counted in yesterdays date, and remaining 3 counted in todays date (as my config is at UTC+3)

**Comment by pedrogaldino** [comment]
Same issue here. Using nova 3.6.0

**Comment by bcollins-oxford** [comment]
I am using UTC as set in app.timezone but still it's not returning the correct number of results.

**Comment by AlbinoDrought** [comment]
Still exists on latest

**Comment by qwertywelch** [comment]
New to Nova, would love to see this issue addressed. Issue still present on 3.8.2. 

Screenshot of a `maxByHours` trend on a table which has rows added every minute with an integer value of how many users were active in the past 15 minutes. The number has never reached zero (in the table's history, and certainly not in recent months). 

Timezone is `America/Chicago (CST or UTC-6), set in `NovaServiceProvider` and in Laravel's `config/app.php`, on the web server, and in MySQL connection and server config. My local comp's timezone is EST or UTC-5, though I've gone through every effort to tell Nova and Laravel to use `America/Chicago`. Tested with multiple timezone configurations but still see similar incorrect behavior. 

Screenshot taken after 1PM Jul 25. I'm no timezone expert, though I'm guessing the timezone UTC offset is what's making Jul 25 9AM-1PM appear as 0 (despite those hours definitely not having a maximum of 0), though what's even more confusing is why Jul 24 9AM - 1PM appear at the end of the chart, instead of chronologically at the beginning.

Thanks to the maintainers for looking into this 👍 

![Screen Shot 2020-07-25 at 2 47 00 PM](https://user-images.githubusercontent.com/1254417/88464278-6cd49780-ce87-11ea-9374-fd589bb3febb.png)

**Comment by ragingdave** [comment]
I'm not sure why this keeps getting closed or why the stale state is being added so quickly, but there definitely is an issue with the timezone handling for metrics. I've run into a similar issue but it seems reversed to those mentioned here.

My app and DB are in UTC, but Nova is displaying in America/New_York. The metric ranges uses the non UTC timeframes when handling at least Value metrics, resulting in some entries that it should be counting for the range being counted and vice versa at the beginning of the range to not be.

As a concrete Query example:
Given a Today Oct 1 at 2:57:33 EST, the 2 queries used to generate the metric are the following:
```
select
  count(`transfer_line_tasks`.`id`) as aggregate
from
  `transfer_line_tasks`
where
  `created_at` between '2020-10-01 00:00:00'
  and '2020-10-01 14:57:33'
```
and 
```
select
  count(`transfer_line_tasks`.`id`) as aggregate
from
  `transfer_line_tasks`
where
  `created_at` between '2020-09-30 00:00:00'
  and '2020-09-30 14:57:33'
```

My question is why is the database query using a date clause which reflects EST rather than the correct UTC as configured in the app and DB level? The queries should be using the UTC version of 2020-10-01 18:57:33 as that's current time according to the app and DB
Additionally, why is this for example a TODAY range, only getting up until the current time? It's not like yesterday suddenly ended early or something?

**Comment by briavers** [comment]
Having the same issue where there is a jump in time 

![Screenshot from 2020-10-02 10-44-26](https://user-images.githubusercontent.com/26483622/94905023-7b998500-049c-11eb-944b-5fdd6d782fa4.png)

Possibly becuase of a combination of people from different zones visiting and the cache saving certain parts

**Comment by bolechen** [comment]
@crynobone can u have time to look this issue for long time, thx.

**Comment by bolechen** [comment]
```
Nova::userTimezone(fn () => 'UTC');
```
 can tmp fix this.

**Comment by crynobone** [comment]
@bolechen We recently released Laravel Nova 3.18.0 with the following changes:

* Fixed an issue preventing metrics from using the application timezone. laravel/nova#1133

From the latest released, what do you see still isn't working as expected in Laravel Nova. Would be useful if someone can create a new issue describing the actual problem on latest Laravel Nova so I can follow up.
